Source/Renderer/getUniformArray.js

```diff
@@ -0,0 +1,575 @@
+/*global define*/
+define([
+        '../Core/Cartesian2',
+        '../Core/Cartesian3',
+        '../Core/Cartesian4',
+        '../Core/Color',
+        '../Core/defined',
+        '../Core/defineProperties',
+        '../Core/DeveloperError',
+        '../Core/Matrix2',
+        '../Core/Matrix3',
+        '../Core/Matrix4',
+        '../Core/RuntimeError'
+    ], function(
+        Cartesian2,
+        Cartesian3,
+        Cartesian4,
+        Color,
+        defined,
+        defineProperties,
+        DeveloperError,
+        Matrix2,
+        Matrix3,
+        Matrix4,
+        RuntimeError) {
+    "use strict";
+    /*global console*/
+
+    /**
+     * @private
+     */
+    var getUniformArray = function(gl, activeUniform, uniformName, locations) {
```

Comment:
would `createUniformArray` be a better name for this function?  `get` usually indicates the thing you're getting already exists.